acceptancetests/assess_updateseries.py

```diff
@@ -0,0 +1,373 @@
+    unit_num = client.get_status().get_service_unit_count(charm_app)
+    client.juju('add-unit', (charm_app))
+    client.wait_for_started()
+    upgrade_release(client, unit_num, args.from_series, args.to_series,
+                    'before')
+
+
+def upgrade_release(client, unit_num, from_series, to_series, reboot):
+    """ Update the series of the given unit.
+
+    :param client: Juju client
+    :param unit_num: unit number to use
+    :param from_series: original series of unit
+    :param to_series: series to update the unit to
+    :param reboot: 'before' or 'after' juju-updateseries is called
+    :return: None
+    :raises JujuAssertionError: if juju-updateseries doesn't act as expected.
+    :raises AssertionError: non juju failures in the process.
+    """
+    unit = "{}/{}".format(charm_app, unit_num)
+    try:
+        output = client.get_juju_output(
+            'ssh', unit, 'sudo do-release-upgrade -f ' +
+            'DistUpgradeViewNonInteractive', timeout=3600)
+    except subprocess.CalledProcessError as e:
+        raise AssertionError(
+            "do-release-upgrade failed on {}: {}".format(unit, e))
+
+    log.info("do-release-upgrade response: ".format(output))
+
+    if reboot in 'before':
+        reboot_unit(client, unit, False)
+
+    # NOTE: attempts to use calls such as client.juju() or remote.ssh() from
+    # this point on return timeouts.  The reboot above timesout, though a
+    # reboot does occur.  Subsequent attempts to access the unit time out.
+    # The cause is unknown, but suspected to be in the CI packages as
+    # subprocess works and is used from here on out.
+
+    assert_correct_series(client, unit, to_series)
+
+    cmd = build_ssh_cmd(
+            client, unit, 'sudo juju-updateseries --to-series ' +
+            '{} --from-series {}'.format(to_series, from_series))
+    try:
+        output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        raise JujuAssertionError(
+            "error running juju-updateseries on {}: {}".format(unit, e))
+
+    log.info(
+            "juju-updateseries ... succeeded:\n{}".format(output))
+
+    if "successfully copied tools and relinked agent tools" not in output:
+        raise JujuAssertionError("failure in juju-updateseries")
+
+    cmd = build_ssh_cmd(client, unit,
+                        'sudo juju-updateseries --to-series ' +
+                        '{} --from-series {} --start-agents'.format(
+                            to_series, from_series))
+    try:
+        output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        if reboot in 'before':
+            if expectedStartAgents.format(charm_app, unit_num, unit_num) \
+                    not in output:
+                raise JujuAssertionError("above output incorrect")
+        elif reboot in 'after':
+            log.info("where does the juju ERROR output go?")
```

Comment:
Is it not in e.output? (note output will be empty due to the exception occuring, thus the stderr text should reside in e.output).

Reply:
I was looking in e, can couldn't find it.  Appears its in e.output though.  TY!